File: easycontrol/server/src/main/java/top/saymzx/easycontrol/server/entity/Options.java
/*
 * 本项目大量借鉴学习了开源投屏软件：Scrcpy，在此对该项目表示感谢
 */
package top.saymzx.easycontrol.server.entity;

public final class Options {
  public static int serverPort=25166;
  public static boolean listenerClip=true;
  public static boolean isAudio = true;
  public static int maxSize = 1600;
  public static int maxVideoBit = 4000000;
  public static int maxFps = 60;
  public static boolean keepAwake = true;
  public static boolean supportH265 = true;
  public static boolean supportOpus = true;
  public static String startApp = "";

  public static void parse(String... args) {
    for (String arg : args) {
      int equalIndex = arg.indexOf('=');
      if (equalIndex == -1) throw new IllegalArgumentException("参数格式错误");
      String key = arg.substring(0, equalIndex);
      String value = arg.substring(equalIndex + 1);
      switch (key) {
        case "serverPort":
          serverPort = Integer.parseInt(value);
          break;
        case "listenerClip":
          listenerClip = Integer.parseInt(value) == 1;
          break;
        case "isAudio":
          isAudio = Integer.parseInt(value) == 1;
          break;
        case "maxSize":
          maxSize = Integer.parseInt(value);
          break;
        case "maxFps":
          maxFps = Integer.parseInt(value);
          break;
        case "maxVideoBit":
          maxVideoBit = Integer.parseInt(value) * 1000000;
          break;
        case "keepAwake":
          keepAwake = Integer.parseInt(value) == 1;
          break;
        case "supportH265":
          supportH265 = Integer.parseInt(value) == 1;
          break;
        case "supportOpus":
          supportOpus = Integer.parseInt(value) == 1;
          break;
        case "startApp":
          startApp = value;
          break;
      }
    }
  }
}

File: scrcpy_android/server/src/main/java/top/saymzx/scrcpy/server/entity/Options.kt
package top.saymzx.scrcpy.server.entity

object Options {
  var maxSize = 0
  var videoCodec = ""
  var audioCodec = ""
  var videoBitRate = 0
  var maxFps = 0

  fun parse(vararg args: String) {
    for (arg in args) {
      val equalIndex = arg.indexOf('=')
      require(equalIndex != -1) { "Invalid key=value pair: \"$arg\"" }
      val key = arg.substring(0, equalIndex)
      val value = arg.substring(equalIndex + 1)
      when (key) {
        "video_codec" -> videoCodec = value
        "audio_codec" -> audioCodec = value
        "max_size" -> maxSize = value.toInt() and 7.inv() // multiple of 8
        "video_bit_rate" -> videoBitRate = value.toInt()
        "max_fps" -> maxFps = value.toInt()
      }
    }
  }
}